onnxruntime/python/onnxruntime_inference_collection.py

```diff
@@ -175,6 +175,19 @@ def enable_fallback(self):
         """
         self._enable_fallback = True
 
+    def _validate_input(self, feed_input_names):
```

Comment:
Validation code was here before. The change is to additionally ensure that it takes into account input names, not just by order and count of inputs. I also refactor the existing validation code into _validate_input because they were duplicated at 2 places.